Dima/netflix-roulette/src/Components/Search/Search.tsx

```diff
@@ -36,20 +36,39 @@ export default class Search extends React.Component<ISearchProps> {
     };
   };
 
+  componentDidUpdate(prevProps: ISearchProps) {
+    if(this.props.location !== prevProps.location) {
+      const URLData = parse(this.props.location.search) as {
+        searchBy: string;
+        search: string
+      };
+      let { searchBy, search } = URLData;
+      if(searchBy) {
+        searchBy === IsearchParams.Title
+          ? this.setState({ searchBy: IsearchParams.Title, searchInput: search }) 
+            : this.setState({ searchBy: IsearchParams.Genre, searchInput: search })
+      } else {
+        this.setState({ searchBy: IsearchParams.Title })
+        search && this.setState({ searchInput: search })
+      };
```

Comment:
can be simplified to 
```
this.setState({
  searchBy:  searchBy === IsearchParams.Title ? 
      IsearchParams.Title : IsearchParams.Genre,
  searchInput: search || ''
});
```
P.s. It's bad approach to call this.setState more then once in a row